NewForce-Cohort-8/tabloidfullstack-the-cage-crushers
# Ticket #19 - View a Post's Comments

**Description**

This code adds functionality to the Tabloid Full Stack application so users can see all of the comments that have been added to a post when they are viewing any individual post's details page. A new button has been added that toggles the visibility of the comments for any given post. Also, with our project manager's approval, we have made this comments functionality all happen within the post details page rather than redirecting between a separate/dedicated page for the comments and the post details page. The user experience will be smoother and more intuitive if they can view / hide comments all from the same page where they're viewing the post's details.

**Related Ticket(s)**
https://trello.com/c/hEDUd2zR/19-19-view-a-posts-comments

Ticket #19 - View A Post's Comments

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-cage-crushers)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-viewcomments-ticket19` as one of the available branches
- `git checkout cm-viewcomments-ticket19` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-viewcomments-ticket19`
- `Start TabloidFullStack.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-cage-crushers/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-cage-crushers/blob/main/SQL/02_Tabloid_Seed_Data.sql)

- Click the execute with debugger solid green button to launch the app
- You should be directed to the /login page and prompted to login - enter “[foo@bar.com](mailto:foo@bar.com)” in the email field and click “Login” - this should redirect you back to the home page.
- Click the “Posts” link/option on the nav bar to get to the Post list page (/posts)
- Click the title of the second post featured on the post lists page - which should be "productize turn-key convergence". (Note: if you have added other posts, this post may not be the second post featured on your post lists page. In this case, you can either find the post in the post list manually, or go directly to the url for this page which is /posts/7). This will take you to the details page for the post that has an Id of 7.
- Once you're on the details page for the aforementioned post, you should see a blue "View Comments" button below the post. Click that thing!!!
- When you click it, you should see the comments for this post appear under a bolded headline that says "COMMENTS:", ordered with most recent comments on the top of the list. Here are the comments you should see for this particular post:

Subject: Nulla ac enim.
Comment: Praesent id massa id nisl venenatis lacinia. Aenean sit amet justo. Morbi ut odio.
Posted By: ecornfoot8
Posted On: 7/2/2020

Subject: Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Mauris viverra diam vitae quam.
Comment: Integer tincidunt ante vel ipsum. Praesent blandit lacinia erat. Vestibulum sed magna at nunc commodo placerat.
Posted By: ecornfoot8
Posted On: 6/15/2020

Subject: Morbi vestibulum, velit id pretium iaculis, diam erat fermentum justo, nec condimentum neque sapien placerat ante.
Comment: Suspendisse potenti. In eleifend quam a odio. In hac habitasse platea dictumst.
Posted By: aotton2
Posted On: 6/7/2020

Subject: Nulla justo.
Comment: Cras mi pede, malesuada in, imperdiet et, commodo vulputate, justo. In blandit ultrices enim. Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
Posted By: ryakushkev4
Posted On: 5/4/2020

- The button's text should also change from "View Comments" to "Hide Comments" after clicking.
- If you click the button again when it says "Hide Comments", all of the comments should disappear from view on the page, and the button's text should revert back to "View Comments"
- If you want to verify that these are the correct comments, simply open a new database query in Visual Studio and run this command. The four comments above should be returned. (Note: there may be more comments if you have added more for any reason. The four aforementioned comments should be returned unless you have been deleting comments via SQL in your database for any reason).
```
SELECT * FROM Comment c
LEFT JOIN Post p ON p.Id = c.PostId
WHERE p.id = 7
ORDER BY c.CreateDateTime DESC
```

Checklist
